fixes #1271 - added toolbar-dense prop to <v-app>

jacekkarczmarczyk opened the pull request:
Not sure if this is the best approach, I'm open for discussion

Usage: 
```
<v-app toolbar-dense>
  <v-toolbar dense />
</v-app>
```

ryanmortier commented:
@jacekkarczmarczyk Please see the issue #1271 again. I've updated it with an additional problem with the `dense` toolbar.

jacekkarczmarczyk commented:
@ryanmortier I've added some more styling so it looks now better, however all these combinations of toolbars/footers/drawers clipped permanent persisten etc are not trivial to fix so I'm not sure wether I've considered all possibilities. I've found also some issues with fixed-footer prop, even without my fixes, but that's another story
Anyway I'm attaching Playground.vue for testing different combinations of drawers/toolbars/footers for those who want work on this issue, probably @johnleider as he assigned himself to it

```
<template>
    <v-app :toolbar-dense="toolbar === 'toolbar-dense prop'" :toolbar="toolbar === 'toolbar prop'" :fixed-footer="footer === 'fixed-footer prop'" :footer="footer === 'footer prop'">
        <v-toolbar class="deep-orange" dark :dense="dense">
            <v-btn icon @click.native.stop="drawer = !drawer">
                <v-icon class="white--text">notifications</v-icon>
            </v-btn>
        </v-toolbar>

        <main>
            <v-container fluid>
                <v-layout row wrap xs12 sm6>
                    <v-flex>
                        <div class="headline">v-app</div>
                        <div><v-select :items="['no footer prop', 'footer prop', 'fixed-footer prop']" v-model="footer" label="footer"></v-select></div>
                        <div><v-select :items="['no toolbar prop', 'toolbar prop', 'toolbar-dense prop']" v-model="toolbar" label="toolbar"></v-select></div>
                        <div class="headline">toolbar</div>
                        <div><v-switch label="dense" v-model="dense" :value="true"></v-switch></div>
                    </v-flex>
                    <v-flex>
                        <div class="headline">navigation-drawer</div>
                        <div><v-switch label="persistent" v-model="persistent" :value="true"></v-switch></div>
                        <div><v-switch label="permanent" v-model="permanent" :value="true"></v-switch></div>
                        <div><v-switch label="temporary" v-model="temporary" :value="true"></v-switch></div>
                        <div><v-switch label="mini-variant" v-model="miniVariant" :value="true"></v-switch></div>
                        <div><v-switch label="clipped" v-model="clipped" :value="true"></v-switch></div>
                    </v-flex>
                </v-layout>
            </v-container>
        </main>

        <v-navigation-drawer
                :permanent="permanent"
                right
                :clipped="clipped"
                :persistent="persistent"
                :temporary="temporary"
                v-model="drawer"
                style="background: yellow"
        >
            <v-toolbar>
                <v-toolbar-title>title</v-toolbar-title>
            </v-toolbar>
        </v-navigation-drawer>

        <v-footer class="deep-orange white--text">
        </v-footer>
    </v-app>
</template>

<script>
    export default {
        data() {
            return {
                drawer: true,
                temporary: false,
                persistent: false,
                permanent: false,
                clipped: false,
                miniVariant: false,
                footer: 'no footer prop',
                toolbar: 'no toolbar prop',
                dense: false,
            };
        },

    };
</script>
```

johnleider commented:
As per many discussions, I will be handling this a different way. Thank you for the contribution!
